python/requirements_test.txt

```diff
@@ -14,6 +14,7 @@ msrestazure==0.6.4
 boto3==1.23.10
 # Todo: investigate if we can get rid of this and exchange for ray.cloudpickle
 cloudpickle==2.2.0
+crc32c==2.3
```

Comment:
Since this is only needed for Datasets tests, we might want to specify this in `python/requirements/data_processing/requirements.txt` instead.

Reply:
Will do. For my knowledge, what's the difference between `data_processing/requirements.txt` and `data_processing/requirements_dataset.txt`? And what is the conditions under which these requirements get picked up (do you have to run pip install -r on these manually? are they compiled into a top-level requirements file or script? are these added as dependencies on the pypi package?)

Reply:
Great question! These requirements manifests are primarily used for CI and can be used for keeping your local development environment consistent with CI.  The primary difference between those two files is that the former is installed for both the Datasets tests CI job and the docs tests (e.g. doctests in docstrings, example notebooks, docs code snippets, etc.) CI job, while the latter is only installed for the Datasets tests CI job. So we've been putting most Datasets-only requirements into `data_processing/requirements.txt`, and only things that are not needed (or cannot be installed) for the docs testing CI job in `data_processing/requirements_dataset.txt`.

I've typically defaulted to `data_processing/requirements.txt` until I've run into issues. 😄

Reply:
Here is the main place where they are used, during the "install dependencies" stage of our CI jobs: https://github.com/ray-project/ray/blob/9125ea398d79617eb80429c05c24ca127f3524c4/ci/env/install-dependencies.sh#L393-L399

Reply:
It'd be nice to add the above comment at the top of these two requirement files about the above intended usage @xcharleslin